src/common/k8s-api/__tests__/kube-api.test.ts

```diff
@@ -409,21 +423,220 @@ describe("KubeApi", () => {
+              beforeEach(async () => {
+                getCall = ingressApi.get({
+                  name: "foo1",
+                  namespace: "default",
+                });
+
+                // This is needed because of how JS promises work
+                await flushPromises();
+              });
+
+              it("makes the request to get the resource", () => {
+                expect(fetchMock.mock.lastCall).toMatchObject([
+                  "http://127.0.0.1:9999/api-kube/apis/networking.k8s.io/v1beta1/namespaces/default/ingresses/foo1",
+                  {
+                    headers: {
+                      "content-type": "application/json",
+                    },
+                    method: "get",
+                  },
+                ]);
+              });
+
+              describe("when the request resolves with no data", () => {
+                let result: Ingress | null;
+
+                beforeEach(async () => {
+                  await fetchMock.resolveSpecific(
+                    ["http://127.0.0.1:9999/api-kube/apis/networking.k8s.io/v1beta1/namespaces/default/ingresses/foo1"],
+                    createMockResponseFromString("http://127.0.0.1:9999/api-kube/apis/networking.k8s.io/v1beta1/namespaces/default/ingresses/foo1", JSON.stringify({})),
+                  );
+                  result = await getCall;
+                });
+
+                it("results in the get call resolving to null", () => {
+                  expect(result).toBeNull();
+                });
+              });
+            });
+          });
+
+          describe("when the request resolves with data", () => {
+            let result: Ingress | null;
+
+            beforeEach(async () => {
+              await fetchMock.resolveSpecific(
+                ["http://127.0.0.1:9999/api-kube/apis/networking.k8s.io/v1beta1/namespaces/default/ingresses/foo"],
+                createMockResponseFromString("http://127.0.0.1:9999/api-kube/apis/networking.k8s.io/v1beta1/namespaces/default/ingresses/foo", JSON.stringify({
+                  apiVersion: "v1",
+                  kind: "Ingress",
+                  metadata: {
+                    name: "foo",
+                    namespace: "default",
+                    resourceVersion: "1",
+                    uid: "12345",
+                  },
+                })),
+              );
+              result = await getCall;
+            });
+
+            it("results in the get call resolving to an instance", () => {
+              expect(result).toBeInstanceOf(Ingress);
+            });
+
+            describe("on the second call to IngressApi.get()", () => {
+              let getCall: Promise<Ingress | null>;
+
+              beforeEach(async () => {
+                getCall = ingressApi.get({
+                  name: "foo1",
+                  namespace: "default",
+                });
+
+                // This is needed because of how JS promises work
+                await flushPromises();
```

Comment:
`flushPromises` is generally needed when the actual promise just about to be resolved isn't accessible in our scope. Most of the time this translates to a design error from an avoidable orphan promise. Not sure if this is the case here.

Reply:
I think this is still required, will check.

Reply:
No this is required because without it none of the calls will actually happen between the `beforeEach` and the `it`.